engine/servermaster/executor_manager.go

```diff
@@ -91,19 +99,26 @@ func NewExecutorManagerImpl(initHeartbeatTTL, keepAliveInterval time.Duration, c
 	}
 }
 
-func (e *ExecutorManagerImpl) removeExecutorImpl(id model.ExecutorID) error {
-	e.mu.Lock()
-	defer e.mu.Unlock()
+// removeExecutorLocked removes an executor from the manager.
+// Note that this method must be called with the lock held.
+func (e *ExecutorManagerImpl) removeExecutorLocked(id model.ExecutorID) error {
 	log.Info("begin to remove executor", zap.String("id", string(id)))
 	exec, ok := e.executors[id]
 	if !ok {
 		// This executor has been removed
 		return errors.ErrUnknownExecutorID.GenWithStackByArgs(id)
 	}
-	addr := exec.Addr
+	addr := exec.Address
 	delete(e.executors, id)
 	e.rescMgr.Unregister(id)
 	log.Info("notify to offline exec")
+
+	ctx, cancel := context.WithTimeout(context.Background(), time.Second*5)
+	defer cancel()
+	if err := e.metaClient.DeleteExecutor(ctx, id); err != nil {
+		return perrors.Trace(err)
```

Comment:
What is the expectation here if an error is returned? Note that `e.notifier.Notify` will not be called, so it seems like the status of  the `ExecutorManagerImpl` has become inconsistent.

Reply:
Yes, it's possible to become inconsistent. Can we tolerate that an executor is removed from memory, but failed to delete from the database? If yes, I think I can move this operation to the last.

Reply:
I move this operation to a goroutine. Since we use ttl mechanism to manage the executor's life cycle, I think we can tolerate that a tombstone executor may be left in the database.

Addressed in https://github.com/pingcap/tiflow/commit/e5d1fc51f98a045215e3edfd1a67a3efe0e93733.